core/src/main/scala/scala/pickling/internal/DefaultPicklerRegistry.scala

```diff
@@ -52,7 +52,15 @@ final class DefaultPicklerRegistry(generator: RuntimePicklerGenerator) extends P
   override def registerUnpickler[T](key: String, p: Unpickler[T]): Unit =
     unpicklerMap.put(key, p)
 
-  /** Checks the existince of an unpickler. */
+  /** Checks the existence of a pickler ignoring the registered generators. */
+  override def lookupExistingPickler(key: String): Option[Pickler[_]] =
```

Comment:
Yes, I know they do not involve runtime reflection. But we want that a concrete pickler has preference over the generator. That is to say that `Pickler[List[String]]` should be returned and registered instead of returning a generator `Pickler[List[_]]`. That's why these methods exist: we want to check when we initialize concrete (un)picklers that there is no *other* concrete (un)pickler for that type. This was done with #410 on mind, thus those tests will pass.